ethereum_history_api/circuits/lib/src/misc/arrays.nr

```diff
@@ -22,9 +22,16 @@ pub fn resize<TItem, NEW_LEN, OLD_LEN>(src: [TItem; OLD_LEN]) -> [TItem; NEW_LEN
 /// * `src` - Source array
 /// * `offset` - Offset in source array 
 pub(crate) fn memcpy<N, M>(dest: &mut [u8; N], src: [u8; M], offset: u64) {
-    assert(offset + N <= M, "source index out of bound");
+    memcpy_up_to_length(dest, src, offset, N);
+}
+
+pub(crate) fn memcpy_up_to_length<N, M>(dest: &mut [u8; N], src: [u8; M], offset: u64, length: u64) {
```

Comment:
btw memcpy seems not very idiomatic, should we look for better names, i.e.:
```rust
copy_array_into(dest: &mut [u8; N], src: [u8; M], offset: u64)
copy_subarray_into(dest: &mut [u8; N], src: [u8; M], offset: u64, length: u64)
```

@LogvinovLeon WDYT?

Reply:
Thought about it before, but didn't find a good solution.

`memcpy` is idiomatic, but in a C-like languages where you can have both array & offset expressed as a memory pointer,
Then `memcpy(&src, &dest, len)` makes sense.


From what I've seen in code - we need three functions:
Copy whole array into dest starting from 0,
Copy whole array into dest starting from offset,
Copy part of array into dest starting from 0,

But maybe more will be needed later.
Unfortunately - we can't use the default args like in other languages.